testing/docker/etcd_setup.go

```diff
@@ -101,11 +126,30 @@ func removeEtcdContainer(client *client.Client, testName string) {
 	client.ContainerRemove(context.Background(), containerName, types.ContainerRemoveOptions{Force: true})
 }
 
-func waitForEtcdToBeReady(client *client.Client, id string) (string, error) {
-	inspect, err := client.ContainerInspect(context.Background(), id)
-	if err != nil {
-		return "", err
+func waitForEtcdToBeReady(client *client.Client, id string, grpcPort, metricsPort int) (string, error) {
+	healthURL := fmt.Sprintf("http://localhost:%d/health", metricsPort)
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+
+	for {
+		select {
+		case <-ctx.Done():
+			cancel()
+			return "", errors.New("failed to start container in a timely manner")
+		case <-time.After(1*time.Second):
```

Comment:
Oh. This is nice. I didn't know about .After

Reply:
I've been learning 😉